docs/source/Library.md

```diff
@@ -49,163 +55,150 @@ model = onmt.models.model.NMTModel(encoder, decoder)
-    Epoch  0,     4/   10; acc:  22.68; ppl: 387.03; 1511 src tok/s; 1462 tgt tok/s; 1514090454 s elapsed
-    Epoch  0,     5/   10; acc:  24.58; ppl: 345.44; 1625 src tok/s; 1509 tgt tok/s; 1514090454 s elapsed
-    Epoch  0,     6/   10; acc:  25.37; ppl: 314.39; 1586 src tok/s; 1493 tgt tok/s; 1514090454 s elapsed
-    Epoch  0,     7/   10; acc:  26.14; ppl: 291.15; 1593 src tok/s; 1520 tgt tok/s; 1514090455 s elapsed
-    Epoch  0,     8/   10; acc:  26.32; ppl: 274.79; 1606 src tok/s; 1545 tgt tok/s; 1514090455 s elapsed
-    Epoch  0,     9/   10; acc:  26.83; ppl: 247.32; 1669 src tok/s; 1614 tgt tok/s; 1514090455 s elapsed
-    Validation
-    Epoch  0,    11/   10; acc:  13.41; ppl: 111.94;   0 src tok/s; 7329 tgt tok/s; 1514090464 s elapsed
-    Epoch  1,     0/   10; acc:   6.59; ppl: 147.05; 1849 src tok/s; 1743 tgt tok/s; 1514090464 s elapsed
-    Epoch  1,     1/   10; acc:  22.10; ppl: 130.66; 2002 src tok/s; 1957 tgt tok/s; 1514090464 s elapsed
-    Epoch  1,     2/   10; acc:  20.16; ppl: 122.49; 1748 src tok/s; 1760 tgt tok/s; 1514090464 s elapsed
-    Epoch  1,     3/   10; acc:  23.52; ppl: 117.41; 1690 src tok/s; 1698 tgt tok/s; 1514090464 s elapsed
-    Epoch  1,     4/   10; acc:  24.16; ppl: 119.42; 1647 src tok/s; 1662 tgt tok/s; 1514090464 s elapsed
-    Epoch  1,     5/   10; acc:  25.44; ppl: 115.31; 1775 src tok/s; 1709 tgt tok/s; 1514090465 s elapsed
-    Epoch  1,     6/   10; acc:  24.05; ppl: 115.11; 1780 src tok/s; 1718 tgt tok/s; 1514090465 s elapsed
-    Epoch  1,     7/   10; acc:  25.32; ppl: 109.59; 1799 src tok/s; 1765 tgt tok/s; 1514090465 s elapsed
-    Epoch  1,     8/   10; acc:  25.14; ppl: 108.16; 1771 src tok/s; 1734 tgt tok/s; 1514090465 s elapsed
-    Epoch  1,     9/   10; acc:  25.58; ppl: 107.13; 1817 src tok/s; 1757 tgt tok/s; 1514090465 s elapsed
-    Validation
-    Epoch  1,    11/   10; acc:  19.58; ppl:  88.09;   0 src tok/s; 7371 tgt tok/s; 1514090474 s elapsed
-
-
-To use the model, we need to load up the translation functions
+To use the model, we need to load up the translation functions. A Translator object requires the vocab fields, a global scorer, general options and model options, here default values collected with dummy parsers.
 
 
 ```python
 import onmt.translate
-```
+import configargparse
 
+dummy_parser = configargparse.ArgumentParser()
+onmt.opts.model_opts(dummy_parser)
+model_opt = dummy_parser.parse_known_args([])[0]
 
-```python
-translator = onmt.translate.Translator(beam_size=10, fields=data.fields, model=model)
-builder = onmt.translate.TranslationBuilder(data=valid_data, fields=data.fields)
+dummy_parser = configargparse.ArgumentParser()
+onmt.opts.translate_opts(dummy_parser)
+opt = dummy_parser.parse_args("-model dummymodel -src dummysrc")
+
+scorer = onmt.translate.GNMTGlobalScorer.from_opt(opt)
+translator = onmt.translate.Translator(model=model, fields=vocab_fields, opt=opt, model_opt=model_opt, global_scorer=scorer)
```

Comment:
These are a bit "hacky" but I couldn't create a Translator object without opt and model_opt.

Reply:
FYI this is something I want to fix. Translator ideally would have a from_opt constructor and a regular constructor so that library users don't have to do stuff like this.

Reply:
Integrated, thanks!